### ISSUE TITLE
source-built 8.0 SDK fails to resolve source-built .net7.0 apphost

### ISSUE THREAD

**Issue by tmds** [dotnet/source-build#3421]
I've installed the source-built .NET 8.0 preview 3 packages on Fedora side-by-side with the .NET 7.0 packages.

My existing `net7.0` projects no longer compile: instead of finding back the .NET 7.0 apphost, they try and look for it on nuget.org.

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

</Project>
``` 

```sh
$ dotnet build
MSBuild version 17.6.0-preview-23174-01+e7de13307 for .NET
  Determining projects to restore...
/tmp/console/console.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Host.fedora.37-x64. No packages exist with this id in source(s): nuget.org
  Failed to restore /tmp/console/console.csproj (in 840 ms).

Build FAILED.

/tmp/console/console.csproj : error NU1101: Unable to find package Microsoft.NETCore.App.Host.fedora.37-x64. No packages exist with this id in source(s): nuget.org
    0 Warning(s)
    1 Error(s)
```

The `Microsoft.NETCoreSdk.BundledVersions.props` for the .NET 8 SDK includes:
```xml
...
    <KnownAppHostPack Include="Microsoft.NETCore.App"
                      TargetFramework="net7.0"
                      AppHostPackNamePattern="Microsoft.NETCore.App.Host.**RID**"
                      AppHostPackVersion="7.0.2"
                      AppHostRuntimeIdentifiers="...;fedora.37-x64"
                      ExcludedRuntimeIdentifiers="android"
                      />
...
```

The issue is the `AppHostPackVersion`: it doesn't match with the `7.0.4` version installed:
```sh
$ ls /usr/lib64/dotnet/packs/Microsoft.NETCore.App.Host.fedora.37-x64
6.0.15  7.0.4  8.0.0-preview.3.23174.8
```

I think the main thing keeping these versions in sync is them being released in tandem?

The .NET 8.0 preview 3 was released about 2 weeks ago, at about the same time as .NET 7.0.5.
The preview 3 is referring to an older .NET 7 version.
Even if it were referring to the latest (7.0.5), that would still not match with what we have available in Fedora.

If we can't really keep these things in sync, maybe we should consider removing the host rid from the `AppHostRuntimeIdentifiers` of older .NET versions?
Then when you publish for `net7.0` on a .NET 8 SDK, you use the Microsoft `linux-x64` Host instead of the `fedora.37-x64`.

cc @MichaelSimons @dsplaisted @omajid

**Comment by dsplaisted** [comment]
I don't think we can generally keep these in sync during previews.  For GA releases I believe we do keep these in sync.

@marcpopMSFT

**Comment by marcpopMSFT** [comment]
Correct. We don't typically have the current public branches target the unreleased down level versions as otherwise you can't use those daily builds and it also impacts testing. We try to stay at N-1 though we fell behind for preview 3: https://github.com/dotnet/installer/blob/release/8.0.1xx-preview3/eng/Versions.props#L26  We've fixed that for preview 4 which should be targeting 7.0.5 which is the released version at the time we snapped the branch.

**Comment by tmds** [comment]
> We've fixed that for preview 4 which should be targeting 7.0.5 which is the released version at the time we snapped the branch.

@marcpopMSFT won't there be a 7.0.6 at that time?

@omajid what's the reason we haven't built the 7.0.5 packages for Fedora?

**Comment by omajid** [comment]
> what's the reason we haven't built the 7.0.5 packages for Fedora?

I have built them all. You can see the updates going out here: https://bodhi.fedoraproject.org/updates/?packages=dotnet7.0

```
$ sudo dnf info dotnet-sdk-7.0
[sudo] password for omajid: 
Last metadata expiration check: 0:43:04 ago on Thu 27 Apr 2023 05:37:17 AM.
Installed Packages
Name         : dotnet-sdk-7.0
Version      : 7.0.105
Release      : 1.fc38
Architecture : x86_64
Size         : 299 M
Source       : dotnet7.0-7.0.105-1.fc38.src.rpm
...
$ dotnet --info
.NET SDK:
 Version:   7.0.105
 Commit:    e1bc5e001c

Runtime Environment:
 OS Name:     fedora
 OS Version:  38
 OS Platform: Linux
 RID:         fedora.38-x64
 Base Path:   /usr/lib64/dotnet/sdk/7.0.105/
...
```

**Comment by tmds** [comment]
> I have built them all. You can see the updates going out here: https://bodhi.fedoraproject.org/updates/?packages=dotnet7.0

My mistake, I ran `dnf install` rather than `dnf update`.

**Comment by marcpopMSFT** [comment]
7.0.6 will be available by the time preview 4 ships but not until then so we wouldn't be able to use the daily builds or easily test ahead so we always fall one month behind for the "implicit version" as we call it.

**Comment by tmds** [comment]
>  so we always fall one month behind

You mean: for preview releases?
Once we're at GA, the .NET 8 releases will point to the .NET 7 releases they are updated with?

That means the issue I'm seeing here is expected to happen until the .NET 8 GA?

**Comment by tmds** [comment]
@marcpopMSFT can you answer the questions from my previous comment?

I'm ok closing the issue if this behavior is as expected during the preview stage.

**Comment by marcpopMSFT** [comment]
Correct, this is only for previews. At the moment, 8.0.1xx and 7.0.4xx are building and shipping out of the public release\ branch. These target N-1 targeting packs to allow customers to pick up the daily builds and use them for testing without needing access to pre-release feeds (which even for first parties can be a pain sometimes).

For 8.0.1xx, we will move to internal with RC1. At that point, we will line up with the shipping version of the downlevel runtime.  You can see we did this for 7.0.1xx-rc1 as well: https://github.com/dotnet/installer/pull/14423.  The two versions in that PR were the downlevel runtime versions that shipped on the same day as rc1.

So confirming that this is temporary through preview 7.

**Comment by MichaelSimons** [comment]
[Triage] Closing because this is the expected behavior until preview 7.
